xdsl/interpreters/pdl_interp.py

```diff
@@ -329,21 +367,15 @@ def run_create_operation(
     def call_func(
         self, interpreter: Interpreter, op: pdl_interp.FuncOp, args: tuple[Any, ...]
     ):
+        if self._rewriter is None:
+            raise InterpretationError(
+                "Expected an active rewriter when calling a pdl_interp function."
+            )
         if op.sym_name.data == "matcher":
-            if self._rewriter is not None:
-                raise InterpretationError(
-                    "Cannot call matcher with an active rewriter."
-                )
-            assert self._rewriter is None
             assert len(args) == 1
             root_op = args[0]
             assert isinstance(root_op, Operation)
-            self.rewriter = PatternRewriter(root_op)
-        else:
-            if self._rewriter is None:
-                raise InterpretationError(
-                    "Expected an active rewriter when calling a rewrite routine."
-                )
+            self.rewriter.current_operation = root_op
```

Comment:
I changed this, the rewriter now needs to be present when a pdl-interp function is called.

Reply:
Worth a separate PR IMO

Reply:
https://github.com/xdslproject/xdsl/pull/4287